harvester_e2e_tests/integration/test_images.py

```diff
@@ -0,0 +1,212 @@
+    testimages.test_get(api_client, 'test-file-upload')
+
+
+@pytest.mark.images_p1
+@pytest.mark.dependency(depends=["create_image_url", "create_image_file"])
+def test_delete_image_recreate(api_client, iso_image_url):
+    """
+    Test create raw and iso type image from file
+
+    Steps:
+    1. Check the image created by URL exists
+    2. Delete the newly created image
+    3. Create an iso file type image from URL
+    4. Check the iso image exists
+    5. Upload an qcow2 file type image
+    5. Delete the newly uploaded file
+    6. Upload a new qcow2 file type image
+    """
+    testimages = test_images.TestImages()
+
+    testimages.test_get(api_client, 'k3os')
+    testimages.test_delete(api_client, 'k3os', 15)
+
+    test_images.create_image_url(api_client, 'k3os', iso_image_url, 600)
+
+    testimages.test_get(api_client, 'test-file-upload')
+    testimages.test_delete(api_client, 'test-file-upload', 15)
+
+    testimages.test_upload_file(api_client, 'test-file-upload', 'test-upload.qcow2', '10M')
+
+
+@pytest.mark.images_p1
+def test_create_invalid_url(api_client):
+    """
+    Test create image from invalid url
+
+    Steps:
+    1. Create image with invalid URL. e.g. - https://test.img
+    2. Check the Image state show as Failed
+    """
+    test_images.test_create_with_invalid_url(api_client, 'test')
+
+
+@pytest.mark.images_p1
+def test_create_invalid_file(api_client):
+    """
+    Test create upload image from invalid file type
+
+    Steps:
+    1. Try to upload invalid image file to images page, something like dmg, or tar.gzps
+    2. Check should get an error
+    """
+
+    os.system('''
+        cat <<EOF >>test-invalid.tar.gzps
+        test
+        EOF
+    ''')
+    resp = api_client.images.create_by_file('test-invalid-file', 'test-invalid.tar.gzps')
+
+    assert 500 == resp.status_code, (
+        f"Failed to upload unsupported file type:{resp.status_code}, {resp.content}"
+    )
+
+    os.system('rm test-img.tar.gzps')
+
+
+@pytest.mark.images_p1
+@pytest.mark.dependency(name="edit_image_in_use")
+def test_edit_image_in_use(api_client, qcow2_image_url, iso_image_url):
+    """
+    Test can edit image which already in use
+
+    Steps:
+    1. Check the image created from URL exists
+    2. Create a VM and use the existing image created from url
+    3. Update the image labels and description
+    4. Check can change the image content
+    """
+    testimages = test_images.TestImages()
+
+    testimages.test_get(api_client, 'k3os')
+
+    spec = api_client.vms.Spec(1, 1)
+    spec.add_image('k3os', "default/k3os")
+    code, data = api_client.vms.create('test-k3os', spec)
+
+    assert 201 == code, (f"Failed to create vm with error: {code}, {data}")
+
+    updates = {
+        "labels": {
+            "usage-label": "yes"
+        },
+        "annotations": {
+            "field.cattle.io/description": 'edit image in use'
+        },
+
+    }
+
+    testimages.test_update_by_input(api_client, 'k3os', updates)
```

Comment:
If you create the test `test_update_by_input` is for the case, then you should not add it to API tests.

Reply:
Move test_update_by_input from `apis/test_images` to `integration/test_images` and wrap in `TestBackendImages` class